src/core/store/modules/notifications/notifications.actions.ts

```diff
@@ -0,0 +1,37 @@
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { ThunkAPI } from '@frameworks/redux';
+
+import { Message } from '@types';
+import notificationMessages from '../../../../client/data/notificationMessages.json';
+
+const dismissMessage = createAction<number>('notifications/dismissMessage');
+
+/* -------------------------------------------------------------------------- */
+/*                                 Clear State                                */
+/* -------------------------------------------------------------------------- */
+
+const clearMessages = createAction<void>('notifications/clearMessages');
+const clearDismissedMessageIds = createAction<void>('notifications/clearDismissedMessageIds');
+
+/* -------------------------------------------------------------------------- */
+/*                                 Fetch Data                                 */
+/* -------------------------------------------------------------------------- */
+const importedNotifications = notificationMessages as Message[];
+
+const getNotificationMessages = createAsyncThunk<Message[], void, ThunkAPI>(
+  'notifications/getNotificationMessages',
+  async () => {
+    return importedNotifications;
```

Comment:
Probably here we should also clear on dismissedMessagesId's the ids that are no longer active/existing, else the state could grow endlessly until a user clears the app local storage and it could produce perf issues for that user

Reply:
Lets filter them on the reducer, and just add the active messages to state. Still dont compare with user dismissed messages though, since you do that already on selector. Intention should be this list will be short anyways, and we can always trim it.

Reply:
Moved the filtering of active messages from notifications.selectors.ts to notifications.reducer.ts

Reply:
this will change once we know where we get the notifications from right?

Reply:
Yes, we need a decentralized source to fetch them from